src/hooks.ts

```diff
@@ -62,62 +46,81 @@ interface UseEllipsePositionConfig {
 }
 
 export function useEllipsePosition(props: UseEllipsePositionConfig) {
-  const { axisContext } = usePlotContext();
+  const { axisContext, width, height } = usePlotContext();
   const [xScale, yScale] = validateAxis(axisContext, 'x', 'y');
   const { cx, cy, rx, ry } = props;
 
   return {
-    cx: convertValue(cx, xScale),
-    cy: convertValue(cy, yScale),
-    rx: convertValueAbs(rx, xScale),
-    ry: convertValueAbs(ry, yScale),
+    cx: convertValue(cx, width, xScale),
+    cy: convertValue(cy, height, yScale),
+    rx: convertValueAbs(rx, width, xScale),
+    ry: convertValueAbs(ry, height, yScale),
   };
 }
 
+function ConvertString(value: string, total: number) {
```

Comment:
our convention is to write regular functions camelCase

Reply:
yes okey

Reply:
it is not resolved

Reply:
fixed